src/main/java/org/jenkinsci/plugins/argusnotifier/TagFactory.java

```diff
@@ -34,15 +32,21 @@ private TagFactory() {
      * @return
      */
     static Map<String, String> buildStatusTags(Jenkins jenkins, String projectName) {
-        return ImmutableMap.<String, String>builder()
-                .putAll(hostTag(jenkins))
-                .put(PROJECT.lower(), InvalidCharSwap.swapWithDash(projectName))
-                .build();
+        return ImmutableMap.<String, String>builder().putAll(hostTag(jenkins))
+                .put(PROJECT.lower(), InvalidCharSwap.swapWithDash(projectName)).build();
+    }
+
+    static Map<String, String> buildExtendedStatusTags(Jenkins jenkins, String projectName, String buildNumber,
```

Comment:
As mentioned previously, perhaps this is just one method with an optional `commitId`.